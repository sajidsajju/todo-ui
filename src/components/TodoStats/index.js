import React from "react";
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles({
  main: {
    width: "40em",
    margin: "auto",
    background: "#654321",
    marginBottom: "5em",
  },
  wrapper: {
    paddingLeft: "10px",
    fontSize: "0.8em",
    lineHeight: "1.5em",
    display: "inline",
    color: "#FFFFFF",
  },
  links: {
    display: "inline",
    float: "right",
    paddingRight: "10em",
  },
  span: {
    padding: "8px",
    cursor: "pointer",
    "&:hover": {
      color: "#A9A9A9",
    },
    "&:active": {
      color: "#00FFFF",
    },
  },
});

function TodoStats(props) {
  const classes = useStyles();
  const { todos, displayTodos, sort } = props;

  const allTodos = () => {
    const newTodos = [...todos];
    sort(newTodos);
  };
  const activeTodos = () => {
    const newTodos = [...todos].filter(todo => todo.completed === false);
    sort(newTodos);
  };

  const completedTodos = () => {
    const newTodos = [...todos].filter(todo => todo.completed !== false);

    sort(newTodos);
  };

  return (
    <div className={classes.main}>
      <div className={classes.wrapper}>
        <span>{displayTodos.length} items left</span>
        <div className={classes.links}>
          <span className={classes.span} onClick={allTodos}>
            All
          </span>
          <span className={classes.span} onClick={activeTodos}>
            Active
          </span>
          <span className={classes.span} onClick={completedTodos}>
            Completed
          </span>
        </div>
      </div>
    </div>
  );
}

export default TodoStats;
