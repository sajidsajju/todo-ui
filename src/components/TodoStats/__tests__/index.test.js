import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";

import TodoStats from "../";

describe("<TodoStats  />", () => {
  const todos = [{ text: "hello" }];
  const displayTodos = [{ text: "hello" }];
  const sort = jest.fn();

  it("should match the snapshot", () => {
    const rendered = render(
      <TodoStats todos={todos} displayTodos={displayTodos} sort={sort} />,
    );

    expect(rendered.container.firstChild).toMatchSnapshot();
  });

  it("should match the length of the displayTodos", () => {
    render(<TodoStats todos={todos} displayTodos={displayTodos} sort={sort} />);

    expect(screen.getByText("1 items left")).toBeInTheDocument();
  });

  it("should call allTodos when click event fired on All Button", () => {
    render(<TodoStats todos={todos} displayTodos={displayTodos} sort={sort} />);

    const AllSortButton = screen.getByText("All");
    fireEvent.click(AllSortButton);
    expect(AllSortButton).toBeInTheDocument();
    expect(sort).toHaveBeenCalledTimes(1);
  });

  it("should call activeTodos when click event fired on Active Button", () => {
    render(<TodoStats todos={todos} displayTodos={displayTodos} sort={sort} />);

    const ActiveSortButton = screen.getByText("Active");
    fireEvent.click(ActiveSortButton);
    expect(ActiveSortButton).toBeInTheDocument();
    expect(sort).toHaveBeenCalledTimes(1);
  });

  it("should call completedTodos when click event fired on Completed Button", () => {
    render(<TodoStats todos={todos} displayTodos={displayTodos} sort={sort} />);

    const CompletedSortButton = screen.getByText("Completed");
    fireEvent.click(CompletedSortButton);
    expect(CompletedSortButton).toBeInTheDocument();
    expect(sort).toHaveBeenCalledTimes(1);
  });
});
