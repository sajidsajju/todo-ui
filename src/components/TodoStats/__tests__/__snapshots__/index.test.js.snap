// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoStats  /> should match the snapshot 1`] = `
<div
  class="makeStyles-main-1"
>
  <div
    class="makeStyles-wrapper-2"
  >
    <span>
      1
       items left
    </span>
    <div
      class="makeStyles-links-3"
    >
      <span
        class="makeStyles-span-4"
      >
        All
      </span>
      <span
        class="makeStyles-span-4"
      >
        Active
      </span>
      <span
        class="makeStyles-span-4"
      >
        Completed
      </span>
    </div>
  </div>
</div>
`;
